presto-hive/src/main/java/io/prestosql/plugin/hive/HiveSplitManager.java

```diff
@@ -234,7 +230,6 @@ public ConnectorSplitSource getSplits(
                 directoryLister,
                 executor,
                 concurrency,
-                recursiveDfsWalkerEnabled,
```

Comment:
You can use `getRecursiveDirWalkerEnabled(session)` here instead of removing this parameter

Reply:
I think since we pass in `session` as an argument, it is clearer if we extract this property where it's needed. i.e. the callee doesn't need to worry about whether it needs to override the given `recursiveDfsWalkerEnabled` argument with what's provided in the session. but this is just my opinion.

Reply:
Yes that makes sense, maybe we can set `recursiveDfsWalkerEnabled` once in the constructor of `BackgroundHiveSplitLoader` using the session. Then we won't have to repeatedly look it up from the session during split generation.

Reply:
Thanks, updated as suggested